clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangelogConsumerImpl.java

```diff
@@ -190,12 +297,44 @@ public CompletableFuture<Void> subscribeAll() {
     return this.subscribe(allPartitions);
   }
 
+  public CompletableFuture<Void> internalSeek(Set<Integer> partitions, String targetTopic, SeekFunction seekAction) {
+    return CompletableFuture.supplyAsync(() -> {
+      Set<TopicPartition> topicPartitionSet = new HashSet<>(kafkaConsumer.assignment());
+      // Prune out current subscriptions
+      for (TopicPartition topicPartition: topicPartitionSet) {
+        currentVersionHighWatermarks.remove(topicPartition.partition());
```

Comment:
Can we guarantee here for each integer partition here just mapped to single `TopicPartition` (there will only one version topic for that partition integer)?  Besides, the kakfa consumer here is not safe here, do we need to have some lock protecting those assignment/seek functions?

Reply:
there will only be either one version topic partition or change capture topic partition for a given partition integer.  Thats the intention at least, the user should see a single partition as a continouse logical stream of events.

Admittedly the assignment is a bit frought.  I do fear that with the right combination of subscibtion actions and version swaps we might do something funny here.  I'll add some concurrency blocks on functions which change the assignment.